### PR TITLE
[4.0] Use the injected application and db in the update notification plugin

### PR DESCRIPTION
**Pull request by laoneo**
Use the injected application and db in the update notification plugin.

### CODE REVIEW

File: plugins/system/updatenotification/updatenotification.php [hunk 1]
```diff
@@ -38,6 +37,22 @@
  */
 class PlgSystemUpdatenotification extends CMSPlugin
 {
+	/**
+	 * Application object
+	 *
+	 * @var    \Joomla\CMS\Application\CMSApplication
+	 * @since  3.8.6
```
Comment (SharkyKZ) [inline plugins/system/updatenotification/updatenotification.php]: Use `__DEPLOY_VERSION__` placeholder?

**Comment by SharkyKZ** [comment]
Remove `Factory::getConfig()` here too please:

https://github.com/joomla/joomla-cms/blob/b7f34ade819403faf160f006f6d3e1b278dce691/plugins/system/updatenotification/updatenotification.php#L247-L250

**Comment by laoneo** [comment]
@SharkyKZ good catch, changed it.

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**